File: backend/src/main/java/unpsjb/labprog/backend/model/Pedido.java
package unpsjb.labprog.backend.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.util.Date;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.JoinColumn;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Pedido {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;

    private Date fecha;

    private String observaciones;

    @ManyToOne
    @JoinColumn(name = "domicilio_id")
    private Domicilio domicilio;

    @ManyToOne
    @JoinColumn(name = "cliente_id")
    private Cliente cliente;

}

File: frontend/cli/src/app/pedidos/pedido.ts
import { ArticulosPedido } from "../articulospedido/articulospedido";
import { Cliente } from "../clientes/cliente";
import { Domicilio } from "../domicilios/domicilio";

export interface Pedido {
  id: number;

  fecha: Date;

  observaciones: string;

  domicilio: Domicilio;

  cliente: Cliente;

  total: number;

  completado: boolean;

  articulosPedido: ArticulosPedido[];
}
